apps/comps/components/agent-profile/edit-field.tsx

```diff
@@ -35,10 +35,21 @@ export const EditAgentField: React.FC<EditAgentFieldProps> = ({
   const [localValue, setLocalValue] = useState(value);
   const [inputValue, setInputValue] = useState("");
 
+  // Update localValue when the parent's value prop changes
+  useEffect(() => {
+    setLocalValue(value);
+  }, [value]);
+
   const handleSave = async () => {
-    onSave(inputValue);
-    setLocalValue(inputValue);
-    setDialogOpen(false);
+    try {
```

Comment:
i noticed that when you edit an exiting agent's handle, if the value you input was invalid, you had to refresh the page to properly try with a new value.